syncer/syncer.go

```diff
@@ -944,10 +965,23 @@ func (s *Syncer) addJob(job *job) error {
 			s.tctx.L().Info("meet the second job of a GTID", zap.Any("binlog position", lastLocation))
 		}
 	})
+
+	failpoint.Inject("checkCheckpointInMiddleOfTransaction", func() {
+		if waitXIDStatus(s.waitXIDJob.Load()) == waiting {
+			s.tctx.L().Info("not receive xid job yet", zap.Any("next job", job))
+		}
+	})
+
+	if waitXIDStatus(s.waitXIDJob.Load()) == waitComplete {
```

Comment:
If only check here and there is no subsequent job entry, it will wait for `maxPauseOrStopWaitTime`.

So add a check in `syncer.Run()`.